src/aiida/orm/nodes/data/code/portable.py

```diff
@@ -177,3 +178,31 @@ def filepath_executable(self, value: str) -> None:
             raise ValueError('The `filepath_executable` should not be absolute.')
 
         self.base.attributes.set(self._KEY_ATTRIBUTE_FILEPATH_EXECUTABLE, value)
+
+    @property
+    def filepath_files(self) -> pathlib.PurePosixPath:
+        """Return the absolute path of the directory that contains the code files.
+
+        :return: The directory of the code files required for the `filepath_executable`.
+        """
+        return pathlib.PurePosixPath(self.base.attributes.get(self._KEY_ATTRIBUTE_FILEPATH_FILES))
+
+    @filepath_files.setter
+    def filepath_files(self, value: t.Union[str, pathlib.Path]) -> None:
+        """Set the absolute path of the directory that contains the code files.
+
+        :param value: The absolute filepath of the directory of the uploaded code files.
+        """
+        type_check(value, (str, pathlib.Path))
+
+        if not pathlib.PurePosixPath(value).is_absolute():
+            raise ValueError('`filepath_files` must be an absolute path.')
+
+        if not pathlib.Path(value).exists():
+            raise ValueError('`filepath_files` must be an existing directory.')
```

Comment:
This will also be true for files, so maybe check it is an existing directory with `is_dir`?

Reply:
Fixed!